### PR TITLE
lint: unify Black with Ruff

### PR DESCRIPTION
**Pull request by Borda**
As Ruff [claims](https://pypi.org/project/ruff/0.0.47/#:~:text=ruff%20is%20compatible%20with%20Black,that%20are%20obviated%20by%20autoformatting.) full compatibility with Black, it makes sense just to use one tool for both to prevent eventual collisions; also bumping Ruff to the latest `0.2.2`

cc: @hugovk

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -1,14 +1,11 @@
...
+    rev: v0.2.2
     hooks:
       - id: ruff
-        args: [--fix, --exit-non-zero-on-fix]
-
-  - repo: https://github.com/psf/black-pre-commit-mirror
-    rev: 23.11.0
-    hooks:
-      - id: black
+        args: ["--fix"]
```
Comment (AlexWaygood) [inline .pre-commit-config.yaml:6]: Why did you remove `--exit-non-zero-on-fix` here? :)

**Review by AlexWaygood — COMMENTED** [review]
Thanks! I pushed a few small fixes to your branch. One question:

File: pyproject.toml [hunk 2]
```diff
@@ -56,14 +56,11 @@ packages = ["sphinxlint"]
...
 
-[tool.black]
-
-[tool.ruff]
+[tool.ruff.lint]
 select = [
   "E", # pycodestyle errors
   "F", # pyflakes errors
   "I", # isort
-  "ISC", # flake8-implicit-str-concat
```
Comment (AlexWaygood) [inline pyproject.toml:66]: I removed this since there was a CI warning saying that it might conflict with the formatting hook

File: sphinxlint/checkers.py [hunk 3]
```diff
@@ -445,7 +445,7 @@ def check_line_too_long(file, lines, options=None):
                 continue  # ignore anonymous hyperlink targets
             if _is_very_long_string_literal(line):
                 continue  # ignore a very long literal string
-            yield lno + 1, f"Line too long ({len(line)-1}/{options.max_line_length})"
+            yield lno + 1, f"Line too long ({len(line) - 1}/{options.max_line_length})"
```
Comment (AlexWaygood) [inline sphinxlint/checkers.py:448]: This was reformatted because black can't format inside f-strings yet, whereas ruff can. Black would reformat this too if it could

File: sphinxlint/rst.py [hunk 4]
```diff
@@ -4,6 +4,7 @@
 - All constants are ALL_CAPS
 - All compiled regexes are suffixed by _RE
 """
+
```
Comment (AlexWaygood) [inline sphinxlint/rst.py:7]: Black also does this with the latest version of black; we just had an old version of black pinned in CI

File: .pre-commit-config.yaml (hunk 1 above)
Comment (hugovk) [reply .pre-commit-config.yaml:6]: Should we have `fix` and `exit-non-zero-on-fix` in `pyproject.toml`, or are they (or just one) better here?

Comment (AlexWaygood) [reply .pre-commit-config.yaml:6]: I usually put `--fix` in pyproject.toml, so that it also autofixes if you execute ruff directly.

I don't really know much about the pros and cons of `--exit-non-zero-on-fix`, and whether it should be put in `pyproject.toml` or `.pre-commit-config.yaml`!

Comment (hugovk) [reply .pre-commit-config.yaml:6]: OK, let's just put `--fix` in `pyproject.toml` and leave `--exit-non-zero-on-fix` here :)

**Review by AlexWaygood — APPROVED** [review]

File: .pre-commit-config.yaml
Comment (hugovk) [inline .pre-commit-config.yaml:1]: Whilst we're updating this file, let's also `pre-commit autoupdate --jobs 0`

File: .pre-commit-config.yaml (hunk 1 above)
Comment (Borda) [reply .pre-commit-config.yaml]: Will do...

**Comment by Borda** [comment]
Btw, is there precommit bot enabled?

**Comment by AlexWaygood** [comment]
> Btw, is there precommit bot enabled?

No, we're not using pre-commit.ci right now

File: .pre-commit-config.yaml
Comment (Borda) [reply .pre-commit-config.yaml:1]: Btw, why the jobs argument is needed?

**Comment by Borda** [comment]
> > Btw, is there precommit bot enabled?
> 
> No, we're not using pre-commit.ci right now

Yes, I noticed, just was wondering if there is a reason not to use it?

**Comment by hugovk** [comment]
I'm in favour: it helps fix lint things in PRs, meaning humans can spend more time reviewing non-trivial stuff; and will send a PR (no more than quarterly) to keep the versions up-to-date.

**Comment by AlexWaygood** [comment]
> I'm in favour: it helps fix lint things in PRs, meaning humans can spend more time reviewing non-trivial stuff; and will send a PR (no more than quarterly) to keep the versions up-to-date.

seconded

**Comment by Borda** [comment]
> I'm in favour: it helps fix lint things in PRs, meaning humans can spend more time reviewing non-trivial stuff; and will send a PR (no more than quarterly) to keep the versions up-to-date.

Exactly my experience, now just finding admin who can install/enable https://github.com/marketplace/pre-commit-ci

File: .pre-commit-config.yaml
Comment (hugovk) [reply .pre-commit-config.yaml:1]: You don't need it, it just makes it faster.

Without it updates each in serial. Without, it does them in parallel:

```
  -j JOBS, --jobs JOBS  Number of threads to use. (default 1).
```

(I have a bunch of aliases:)

```console
❯ which pca
pca: aliased to pre-commit autoupdate --jobs 0
❯ which pcr
pcr: aliased to pre-commit run --all-files
❯ which pci
pci: aliased to pre-commit install --allow-missing-config
❯ which pcu
pcu: aliased to pre-commit uninstall
```

Comment (Borda) [reply .pre-commit-config.yaml:1]: nice, will update my macro on local machine :)

Comment (AlexWaygood) [reply .pre-commit-config.yaml:1]: I ran autoupdate in https://github.com/sphinx-contrib/sphinx-lint/pull/110/commits/c825601136112fe86d48aa998b4890f5fd6f0b85

File: .pre-commit-config.yaml [hunk 5]
```diff
@@ -1,14 +1,11 @@
...
       - id: ruff
-        args: [--fix, --exit-non-zero-on-fix]
-
-  - repo: https://github.com/psf/black-pre-commit-mirror
-    rev: 23.11.0
-    hooks:
-      - id: black
+        args: ["--exit-non-zero-on-fix"]
+      - id: ruff-format
+        args: ["--preview"]
```
[the remark this answers is not in the document]
Comment (hugovk) [reply .pre-commit-config.yaml:8]: Yes, if we want to enable unstable formatting, it should be enabled when running via pre-commit and directly on the command line, so we get the same results for both.

**Review by hugovk — APPROVED** [review]
Thanks both!
